lib/iris/pandas.py

```diff
@@ -124,107 +124,100 @@ def _as_pandas_coord(coord):
+    add_aux_coord=False,
+    add_global_attributes=None,
+):
     """
-    Convert a 1D cube to a Pandas Series.
+    Convert a :class:`Cube` to a Pandas `DataFrame`.
 
     Args:
 
-        * cube - The cube to convert to a Pandas Series.
+        * cube:
+            The :class:`Cube` to be converted to a Pandas `DataFrame`.
 
     Kwargs:
 
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data.
+        * copy - Whether to make a copy of the data. Defaults to False.
+        * asmultiindex:
+            If True, returns a `DataFrame` with a `MultiIndex <https://pandas.pydata.org/docs/reference/api/pandas.MultiIndex.html#pandas.MultiIndex>`_.  Defaults to False.
+        * add_aux_coord:
+            If True, add all :class:`Cube` :class:`AuxCoord`s to add to the returned `DataFrame`. Defaults to False.
+        * add_global_attributes:
+            A list of :class:`Cube` attributes to add to the returned `DataFrame`. Defaults to None.
 
     .. note::
 
-        This function will copy your data by default.
-        If you have a large array that cannot be copied,
-        make sure it is not masked and use copy=False.
+        By default, this function seeks to minimise data copying such that the new `DataFrame` shares a reference the `cube.data` values. If you want to break the reference between :class:`Cube` and `DataFrame` select `copy=True`.
 
     """
-    data = cube.data
+    if copy:
+        data = cube.data.copy()
+    else:
+        data = cube.data
     if ma.isMaskedArray(data):
-        if not copy:
-            raise ValueError("Masked arrays must always be copied.")
         data = data.astype("f").filled(np.nan)
-    elif copy:
-        data = data.copy()
-
-    index = None
-    if cube.dim_coords:
-        index = _as_pandas_coord(cube.dim_coords[0])
-
-    series = pandas.Series(data, index)
-    if not copy:
-        _assert_shared(data, series)
-
-    return series
-
-
-def as_data_frame(cube, copy=True):
-    """
-    Convert a 2D cube to a Pandas DataFrame.
-
-    Args:
-
-        * cube - The cube to convert to a Pandas DataFrame.
 
-    Kwargs:
-
-        * copy - Whether to make a copy of the data.
-                 Defaults to True. Must be True for masked data
-                 and some data types (see notes below).
-
-    .. note::
-
-        This function will copy your data by default.
-        If you have a large array that cannot be copied,
-        make sure it is not masked and use copy=False.
-
-    .. note::
+    # Extract dim coord information
+    if cube.ndim != len(cube.dim_coords):
+        # Create dummy dim coord information if dim coords not defined
+        coord_names = ["dim" + str(n) for n in range(cube.ndim)]
+        coords = [range(dim) for dim in cube.shape]
+        for onecoord in cube.dim_coords:
+            for i, dummyc in enumerate(coords):
+                if len(dummyc) == len(onecoord.points):
```

Comment:
Wouldn't this logic cause confusion when a cube has two DimCoords of the same length?

Reply:
Yes, although I originally thought this would be a problem, as the mapping would be the same (wrongly).  So, is there a better way to establish which DimCoord the AuxCoord maps to?

Reply:
I imagine you might want to identify each dim_coord by its dimension. You could try something like:
```
coord_lists = [cube.coords(dimensions=n, dim_coords=True) for n in range(len(cube.ndims))]
coords = [list[0] for list in coord_lists if len(list)>0 else None for list in coord_lists]
```
which should give you a list of DimCoords indexed by their associated dimension.

Reply:
I think [8301caa](https://github.com/SciTools/iris/pull/4669/commits/8301caaedd6b48470136effb708d96e60c93960b) now has better logic